pkg/ddc/thin/transform_fuse.go

```diff
@@ -143,6 +149,70 @@ func (t *ThinEngine) parseFuseImage(runtime *datav1alpha1.ThinRuntime, value *Th
 	}
 }
 
+func (t *ThinEngine) parseHostVolumeFromDataset(dataset *datav1alpha1.Dataset, thinValue *ThinValue) error {
+	index := 0
+	for _, mount := range dataset.Spec.Mounts {
+		for key, value := range mount.Options {
+			if key != common.DatasetOptionFluidFuseHostVolume {
+				continue
+			}
+
+			var hostPath, mountPath string
+			subStrings := strings.Split(value, ":")
+			if len(subStrings) > 2 {
+				return fmt.Errorf("invalid dataset option format, %s: %s", key, value)
+			}
+
+			hostPath = subStrings[0]
+			if !filepath.IsAbs(hostPath) {
+				return fmt.Errorf("invalid dataset option, %s: %s, should be a absolute hostPath", key, value)
+			}
+
+			mountPath = hostPath
+			if len(subStrings) == 2 {
+				mountPath = subStrings[1]
+				if !filepath.IsAbs(mountPath) {
+					return fmt.Errorf("invalid dataset options, %s: %s, should be a absolute mountPath", key, value)
+				}
+
+				for _, volumeMount := range thinValue.Fuse.VolumeMounts {
```

Comment:
The logic for creating corev1.Volume and corev1.VolumeMount is repeated. This can be extracted into a helper function to reduce duplication and improve readability. For example:

```golang
func createVolumeAndMount(index int, hostPath, mountPath string, readOnly bool) (corev1.Volume, corev1.VolumeMount) {
    volumeName := fmt.Sprintf("fluid-fuse-hostvolume-%v", index)
    return corev1.Volume{
        Name: volumeName,
        VolumeSource: corev1.VolumeSource{
            HostPath: &corev1.HostPathVolumeSource{
                Path: hostPath,
            },
        },
    }, corev1.VolumeMount{
        Name:      volumeName,
        MountPath: mountPath,
        ReadOnly:  readOnly,
    }
}
```